core/src/main/kotlin/de/fraunhofer/iem/spha/core/strategy/KpiCalculationStrategy.kt

```diff
@@ -149,4 +152,38 @@ internal abstract class BaseKpiCalculationStrategy : KpiCalculationStrategy {
     }
 
     protected abstract fun internalIsValid(node: KpiNode, strict: Boolean): Boolean
+
+    companion object {
+        fun getResultInValidRange(result: KpiCalculationResult): KpiCalculationResult {
```

Comment:
in strict mode we would fail even earlier as we then enforce the validity of the given KPI hierarchy and a valid hierarchy should not result in invalid values. 
This function should mainly act as a failsafe if we somehow messed up and somewhere get invalid range results.